VSCode/armsnippets.json

```diff
@@ -506,6 +506,145 @@
     ],
     "description": "Availability Set"
 },
+"Azure Firewall": {
+    "prefix": "arm-firewall",
+    "body": [
+        "{",
+        "    \"name\": \"${1:firewall1}\",",
+        "    \"type\": \"Microsoft.Network/azureFirewalls\",",
+        "    \"apiVersion\": \"2019-04-01\",",
+        "    \"location\": \"[resourceGroup().location]\",",
+        "    \"tags\": {},",
+        "    \"properties\": {",
+        "       \"applicationRuleCollections\": [",
+        "           {",
+        "               \"name\": \"${2:appCollectionName1}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${3|100,1-65000|}\",",
+        "                   \"action\": {",
+        "                       \"type\": \"${4|Allow,Deny|}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                       {",
+        "                           \"name\": \"${5:appRuleName1}\",",
+        "                           \"description\": \"${6:appRuleDescription}\",",
+        "                           \"sourceAddresses\": [",
+        "                               \"${7:10.0.0.4}\"",
+        "                           ],",
+        "                           \"protocols\": [",
+        "                              {",
+        "                                  \"protocolType\": \"${8|Http,Https,Mssql|}\",",
+        "                                  \"port\": \"${9|80,443,1433|}\"",
+        "                              }",
+        "                           ],",
+        "                           \"targetFqdns\": [",
+        "                               \"${10|www.microsoft.com,*.microsoft.com|}\"",
+        "                           ],",
+        "                           \"fqdnTags\": []",
+        "                       }",
+        "                   ]",
+        "               }",
+        "           }",
+        "       ],",
+        "       \"natRuleCollections\": [",
+        "           {",
+        "              \"name\": \"${11:natCollectionName1}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${12|100,1-65000|}\",",
```

Comment:
\"priority\": \"${12:100}\",",

Reply:
I have a handful of items with a valid default value, and then the allowed values trailing. Is this not a good approach?

Reply:
From https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax:
"Placeholders can have choices as values. The syntax is a comma-separated enumeration of values, enclosed with the pipe-character, for example ${1|one,two,three|}. When the snippet is inserted and the placeholder selected, choices will prompt the user to pick one of the values."
I don't see any way to provide a description or limitations for a placeholder. In the unlikely case someone provides a value outside the supported range, they'll get a helpful error during template validation, correct?

Reply:
Interesting. I wasn't actually tabbing through, but I do see it now. I'll resolve. I wish there was a way to somehow show allowed values.